File: app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/CreateNoteTable.java
package de.westnordost.streetcomplete.data.osmnotes;

public class CreateNoteTable
{
	public static final String NAME = "osm_create_notes";

	public static class Columns
	{
		public static final String
				ID = "create_id",
				LATITUDE = "latitude",
				LONGITUDE = "longitude",
				TEXT = "text",
				ELEMENT_TYPE = "element_type",
				ELEMENT_ID = "element_id",
				QUEST_TITLE = "quest_title",
				IMAGE_PATHS = "image_paths";
	}
}

File: app/src/main/java/de/westnordost/streetcomplete/data/osmnotes/CreateNoteTable.kt
package de.westnordost.streetcomplete.data.osmnotes

object CreateNoteTable {
    const val NAME = "osm_create_notes"

    object Columns {
        const val ID = "create_id"
        const val LATITUDE = "latitude"
        const val LONGITUDE = "longitude"
        const val TEXT = "text"
        const val ELEMENT_TYPE = "element_type"
        const val ELEMENT_ID = "element_id"
        const val QUEST_TITLE = "quest_title"
        const val IMAGE_PATHS = "image_paths"
    }

    const val CREATE = """
        CREATE TABLE $NAME (
            ${Columns.ID} INTEGER PRIMARY KEY,
            ${Columns.LATITUDE} double NOT NULL,
            ${Columns.LONGITUDE} double NOT NULL,
            ${Columns.ELEMENT_TYPE} varchar(255),
            ${Columns.ELEMENT_ID} int,
            ${Columns.TEXT} text NOT NULL,
            ${Columns.QUEST_TITLE} text,
            ${Columns.IMAGE_PATHS} blob
        );"""
}
